packages/checkout/widgets-sample-app/src/components/ui/on-ramp/onRamp.tsx

```diff
@@ -1,38 +1,86 @@
-import { Checkout, OnRampEventType, WidgetTheme, WidgetType } from "@imtbl/checkout-sdk";
-import { useEffect, useMemo } from "react";
+import {
+  Checkout,
+  OnRampEventType,
+  WidgetTheme,
+  WidgetType,
+} from "@imtbl/checkout-sdk";
+import { useEffect, useMemo, useState } from "react";
 import { WidgetsFactory } from "@imtbl/checkout-widgets";
 
 function OnRampUI() {
-  const checkout = useMemo(() => new Checkout(), [])
-  const onRamp = useMemo(() => new WidgetsFactory(checkout, {}).create(WidgetType.ONRAMP), [checkout])
+  const checkout = useMemo(() => new Checkout(), []);
+  const onRamp = useMemo(
+    () => new WidgetsFactory(checkout, {}).create(WidgetType.ONRAMP),
+    [checkout]
+  );
+  const [skipConnect, setSkipConnect] = useState<boolean>(false);
+  const [toWalletAddress, setToWalletAddress] = useState<string>("");
 
-  const unmount = () => {onRamp.unmount()}
+  const unmount = () => {
+    onRamp.unmount();
+  };
   const mount = () => {
-    onRamp.mount('onramp', {amount: '55', tokenAddress: '0x0000000000000000000000000000000000001010'})}
-  const update = (theme: WidgetTheme) => {onRamp.update({config: {theme}})}
+    onRamp.mount("onramp", {
+      amount: "55",
+      tokenAddress: "0x0000000000000000000000000000000000001010",
+      skipConnect,
+      toWalletAddress,
+    });
+  };
+  const update = (theme: WidgetTheme) => {
+    onRamp.update({ config: { theme } });
+  };
 
   useEffect(() => {
-    mount()
+    // if skipConnect is true, wait until toWalletAddress is set
+    if (skipConnect && !toWalletAddress) return;
+
+    mount();
     onRamp.addListener(OnRampEventType.SUCCESS, (data) => {
-      console.log('SUCCESS', data)
-    })
+      console.log("SUCCESS", data);
+    });
     onRamp.addListener(OnRampEventType.FAILURE, (data) => {
-      console.log('FAILURE', data)
-    })
+      console.log("FAILURE", data);
+    });
     onRamp.addListener(OnRampEventType.CLOSE_WIDGET, () => {
-      unmount()
-    })
-  }, []);
+      unmount();
+    });
+  }, [skipConnect]);
```

Comment:
@jhesgodi does `toWalletAddress` need to be a dependancy here, as we're waiting for it to be set?

Reply:
nice one, thank you!